File: src/Stronghold.Application/Features/Suppliers/SupplierResponse.cs
namespace Stronghold.Application.Features.Suppliers;

public class SupplierResponse
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string? Website { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: stronghold_desktop/lib/features/products/models/supplier_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'supplier_response.freezed.dart';
part 'supplier_response.g.dart';

@freezed
abstract class SupplierResponse with _$SupplierResponse {
  const factory SupplierResponse({
    required int id,
    required String name,
    String? email,
    String? phone,
    String? website,
    required DateTime createdAt,
  }) = _SupplierResponse;

  factory SupplierResponse.fromJson(Map<String, dynamic> json) =>
      _$SupplierResponseFromJson(json);
}

@freezed
abstract class PagedSupplierResponse with _$PagedSupplierResponse {
  const factory PagedSupplierResponse({
    required List<SupplierResponse> items,
    required int totalCount,
    required int totalPages,
    required int currentPage,
    required int pageSize,
  }) = _PagedSupplierResponse;

  factory PagedSupplierResponse.fromJson(Map<String, dynamic> json) =>
      _$PagedSupplierResponseFromJson(json);
}
